src/coreclr/src/vm/eetwain.cpp

```diff
@@ -3063,6 +3067,51 @@ inline unsigned SKIP_LEA_ESP_EBP(int val, PTR_CBYTE base, unsigned offset)
     return(offset + delta);
 }
 
+inline unsigned SKIP_LEA_EAX_ESP(int val, PTR_CBYTE base, unsigned offset)
```

Comment:
The unwinding algorithm is part of R2R contract. This will require bumping the major R2R format version.

Reply:
I never changed the R2R format version in the past. Is there any guideline I should follow to accomplish that?

Reply:
It is about bumping the version number in https://github.com/dotnet/runtime/blob/master/src/coreclr/inc/readytorun.h and other places this file points to.

There are two ways how to evolve the R2R contract:
- Backward compatible: It means keeping both old and new version of the unwinding algorithm, and choosing the right algortihm based on the R2R format version in the given image
- Non-backward compatible (ie breaking): We will fallback to JIT for the older R2R images.
We prefer the backward compatible evolution when we have a choice.

@dotnet/crossgen-contrib Do you have thoughts on whether we want to do R2R version break for 6.0?

Reply:
Actually, the simplest thing to do for this one may be to do support both old and new unwind sequences. The unwinding inspects the instruction stream and so it should be possible to handle both without passing any explicit versions around.

You will still want to change these numbers to:
```
#define READYTORUN_MAJOR_VERSION 0x0005
#define READYTORUN_MINOR_VERSION 0x0001
```
but that will still be backward compatible change.

Reply:
Thank you for suggestion, Jan! I will rework this and add support the old unwind sequence.

Reply:
I am still validating the changes but so far I haven't seen any issues with 3.1 and 5.0 R2R images.

Reply:
Don't forget to also include READYTORUN_MAJOR_VERSION bump that I have mentioned above.

Reply:
Thanks for reminding to do the bump - I updated both readytorun.h and ModuleHeaders.cs.